### DISCUSSION TITLE
Search the Internet and find reference materials on your own

### DISCUSSION THREAD

**Discussion by hellosunghyun** [glowingjade/obsidian-smart-composer#178] [Ideas & Feature Requests]
Use Google Search API or other technology to extract keywords from prompts and retrieve data.

**Comment by glowingjade** [comment]
We’re working on adding a web search feature, similar to the @web functionality in Cursor IDE. Look forward to it!

**Reply by BradKML** [reply]
Seconding this since Obsidian Copilot has limited options https://github.com/logancyang/obsidian-copilot

**Reply by kevin-on** [reply]
@BradKML Thanks for the feedback! We're working on a few things that should help with this:

- Support for Sonar models will be coming soon
- For general web search capabilities, we're currently exploring integrating MCP (Multi-Context Provider), which would allow you to use something like Perplexity MCP with LLM models that support MCP

Does this approach sound good to you, or do you have other ideas we should consider?

Also, I'm curious about the specific limitations you've encountered with obsidian-copilot, as I haven't used it myself. What functionality are you finding most restricted?

**Reply by BradKML** [reply]
@kevin-on key part is in-housing or FOSSing everything including web search tools, there should be equivalents to Perplexity like Perplexica i hope?
